From axel.lin@gmail.com  Wed Oct 20 14:26:37 2010
Subject: USB: imx21-hcd - fix off by one resource size calculation
From: Axel Lin <axel.lin@gmail.com>
Cc: Greg Kroah-Hartman <gregkh@suse.de>,
	Martin Fuzzey <mfuzzey@gmail.com>, linux-usb@vger.kernel.org
Date: Sat, 16 Oct 2010 22:48:08 +0800
Message-ID: <1287240488.2831.1.camel@phoenix>

Signed-off-by: Axel Lin <axel.lin@gmail.com>
Tested-by: Jon Povey <Jon.Povey@racelogic.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/imx21-hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/imx21-hcd.c
+++ b/drivers/usb/host/imx21-hcd.c
@@ -1906,7 +1906,7 @@ failed_clock_set:
 failed_clock_get:
 	iounmap(imx21->regs);
 failed_ioremap:
-	release_mem_region(res->start, res->end - res->start);
+	release_mem_region(res->start, resource_size(res));
 failed_request_mem:
 	remove_debug_files(imx21);
 	usb_put_hcd(hcd);
